sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_internal/async_challenge_auth_policy.py

```diff
@@ -21,35 +21,27 @@
 from .challenge_auth_policy import _enforce_tls, _update_challenge
 
 if TYPE_CHECKING:
-    from typing import Any, Optional
     from azure.core.pipeline import PipelineRequest, PipelineResponse
+    from .http_challenge import HttpChallenge
 
 
 class AsyncChallengeAuthPolicy(AsyncBearerTokenCredentialPolicy):
     """policy for handling HTTP authentication challenges"""
 
-    def __init__(self, *args: "Any", **kwargs: "Any") -> None:
-        self._last_tenant_id = None  # type: Optional[str]
-        super().__init__(*args, **kwargs)
-
     async def on_request(self, request: "PipelineRequest") -> None:
+        _enforce_tls(request)
         challenge = ChallengeCache.get_challenge_for_url(request.http_request.url)
         if challenge:
-            if self._last_tenant_id == challenge.tenant_id:
-                # Super can handle this. Its cached token, if any, probably isn't from a different tenant, and
-                # it knows the scope to request for a new token. Note that if the vault has moved to a new
-                # tenant since our last request for it, this request will fail.
-                await super().on_request(request)
-            else:
-                # acquire a new token because this vault is in a different tenant
-                await self.authorize_request(request, *self._scopes, tenant_id=challenge.tenant_id)
+            # Note that if the vault has moved to a new tenant since our last request for it, this request will fail.
+            await self._handle_challenge(request, challenge)
```

Comment:
This effectively disables token caching on the policy because nothing uses the cached token anymore--`_handle_challenge` calls `authorize_request`, which replaces the cached token with a new one.